rfcs/experimental/flow_graph_wait_for_one/README.md

```diff
@@ -0,0 +1,631 @@
+
+The feature should improve the Flow Graph performance in scenarios where multiple threads submits the work into the
+Flow Graph simultaneously and each of them should
+
+## Introduction
+
+Current API of oneTBB Flow Graph allows doing two basic actions after building the graph:
+
+- Submitting messages in some nodes using the ``node.try_put(msg)`` API.
+- Waiting for completion of **all** messages in ghe graph using the ``graph.wait_for_all()`` API.
+
+Since the only API currently available for waiting until the work would be completed waits for all tasks in the graph
+to complete, there can be negative performance impact in use cases when the thread submitting the work should be notified as soon as possible
+when the work is done. Having only ``graph.wait_for_all()`` forces the thread to wait until **all** of the tasks in the Flow Graph, no meter
+corresponds to the waited message or not.
+
+Consider the following example:
+
+```cpp
+
+struct ComputeInput;
+struct ComputeOutput;
+
+// High-level computation instance that performs some work
+// using the oneTBB Flow Graph under the hood
+// but the Flow Graph details are not expressed as part of public API
+class ComputeTool
+{
+private:
+    oneapi::tbb::flow::graph m_graph;
+    oneapi::tbb::flow::broadcast_node<ComputeInput> m_start_node;
+    // Other Flow Graph nodes
+public:
+    ComputeTool()
+    {
+        // Builds the Flow Graph
+    }
+
+    // Performs the computation using the user-provided input
+    ComputeOutput compute(ComputeInput input)
+    {
+        m_start_node.try_put(input); // Submit work in the graph
+        m_graph.wait_for_all(); // Waiting for input to be processed
+    }
+};
+
+int main()
+{
+    ComputeTool shared_tool;
+
+    oneapi::tbb::parallel_for(0, 10,
+        [&](int i)
+        {
+            // Preparing the input for index i
+
+            ComputeOutput output = shared_tool.compute(input);
+
+            // Post process output for index i
+        });
+}
+
+```
+
+The ``ComputeTool`` is a user interface that performs some computations on top of oneTBB Flow Graph. The function ``compute(input)`` should
+perform the computations for the provided input. Since the Flow Graph is used, it should submit the message into the graph and wait for its
+completion. This function also can be called concurrently from several threads as it shown in the ``main``.
+
+Since the only available API in the Flow Graph is ``wait_for_all()``, each thread submitting the work to the graph would be required to wait
+until **all** of the tasks would be done, no meter if these tasks corresponds to the input processed by this thread or not. If some
+post-processing is required on each thread after receiving the computation result, it would be only possible to start it when the Flow Graph would be completed what can be inefficient if the post-processing of lightweight graph tasks would be blocked by processing the more mature input.
+
```

Comment:
```suggestion
post-processing is required on each thread after receiving the computation result, it would be only safe to start it when the Flow Graph is complete. 
This is inefficient since the post-processing of lightweight graph tasks are blocked by processing of unrelated inputs.

```